src/layout/useLayout.ts

```diff
@@ -200,32 +187,81 @@ export const useLayout = ({
   );
 
   const positionCanvas = useCallback(
-    (position: CanvasPosition) => {
+    (position: CanvasPosition, animated = false) => {
       positionVector(position);
-      positionScroll(position);
+      positionScroll(position, animated);
     },
     [positionScroll, positionVector]
   );
 
-  useEffect(() => {
-    ref?.current?.scrollTo(scrollXY[0], scrollXY[1]);
-  }, [scrollXY, ref]);
-
   useEffect(() => {
     if (scrolled.current && defaultPosition) {
       positionVector(defaultPosition);
     }
   }, [positionVector, zoom, defaultPosition]);
 
-  const fitCanvas = useCallback(() => {
-    if (layout) {
-      const heightZoom = height / layout.height;
-      const widthZoom = width / layout.width;
-      const scale = Math.min(heightZoom, widthZoom, 1);
-      setZoom(scale - 1);
-      positionCanvas(CanvasPosition.CENTER);
-    }
-  }, [height, layout, width, setZoom, positionCanvas]);
+  const fitCanvas = useCallback(
+    (animated = false) => {
+      if (layout) {
+        const heightZoom = height / layout.height;
+        const widthZoom = width / layout.width;
+        const scale = Math.min(heightZoom, widthZoom, 1);
+        setZoom(scale - 1);
+        positionCanvas(CanvasPosition.CENTER, animated);
+      }
+    },
+    [height, layout, width, setZoom, positionCanvas]
+  );
+
+  /**
+   * This centers the chart on the canvas, zooms in to fit the specified node, and scrolls to center the node in the viewport
+   */
+  const fitToNode = useCallback(
+    (nodeId: string, animated = true) => {
+      if (layout && layout.children) {
+        const node = findNode(layout.children, nodeId);
```

Comment:
I would probably extract this next bits of logic into a pure function so you can write a unit test considering what logic you wrote here it seems like a good idea.

Reply:
ok